src/Microsoft.ML.Transforms/UngroupTransform.cs

```diff
@@ -327,11 +327,11 @@ public UngroupBinding(IExceptionContext ectx, Schema inputSchema, UngroupMode mo
                     int col;
                     if (!inputSchema.TryGetColumnIndex(name, out col))
                         throw ectx.ExceptUserArg(nameof(Arguments.Column), "Pivot column '{0}' is not found", name);
-                    var colType = inputSchema[col].Type;
-                    if (!colType.IsVector || !(colType.ItemType is PrimitiveType))
+                    var colType = inputSchema[col].Type as VectorType;
+                    if (colType == null || !(colType.ItemType is PrimitiveType))
```

Comment:
Could probably just be `if (!(colType?.ItemType is PrimitiveType))`? Just a small bit cleaner.

Reply:
Actually, now that I'm thinking about it a little bit more, given that `VectorType.ItemType` *is* in fact a `PrimitiveType`, is this test here equivalent to `colType == null`?

Reply:
Good call. Fixed.